arlas-core/src/main/java/io/arlas/server/core/FluidSearch.java

```diff
@@ -776,45 +773,45 @@ else if (aggregationBuilder instanceof GeoGridAggregationBuilder)
+                        aggregationBuilder.subAggregation(metricAggregation);
+                        break;
+                    case CENTROID:
+                        metricAggregation = AggregationBuilders.geoCentroid(AggregatedGeometryEnum.CENTROID.value() + AGGREGATED_GEOMETRY_SUFFIX).field(aggregationGeoField);
+                        aggregationBuilder.subAggregation(metricAggregation);
+                        break;
                 }
-                // if aggregationModel.type ==  AggregationTypeEnum.geohash then we already return the centroid of each geohash by default => nothing to implement
-            } else {
-                String[] includes = {collectionReference.params.geometryPath, collectionReference.params.centroidPath};
-                String sortField = (aggregationModel.fetchGeometry.field != null) ? aggregationModel.fetchGeometry.field : collectionReference.params.timestampPath;
-                if (aggregationModel.fetchGeometry.strategy == AggregatedGeometryStrategyEnum.first) {
-                    TopHitsAggregationBuilder topHitsAggregationBuilder = AggregationBuilders.topHits(FIRST_GEOMETRY).size(1).sort(sortField, SortOrder.ASC).fetchSource(includes, null);
-                    aggregationBuilder.subAggregation(topHitsAggregationBuilder);
-                } else if (aggregationModel.fetchGeometry.strategy == AggregatedGeometryStrategyEnum.last) {
-                    TopHitsAggregationBuilder topHitsAggregationBuilder = AggregationBuilders.topHits(LAST_GEOMETRY).size(1).sort(sortField, SortOrder.DESC).fetchSource(includes, null);
-                    aggregationBuilder.subAggregation(topHitsAggregationBuilder);
+            });
+        }
+        return aggregationBuilder;
+    }
+
+    private ValuesSourceAggregationBuilder setRawGeometries(Aggregation aggregationModel, ValuesSourceAggregationBuilder aggregationBuilder) throws ArlasException {
+        if (aggregationModel.rawGeometries != null && !aggregationModel.rawGeometries.geometries.isEmpty()) {
+            String[] includes = aggregationModel.rawGeometries.geometries.stream().toArray(String[]::new);
+            TopHitsAggregationBuilder topHitsAggregationBuilder = AggregationBuilders.topHits(RAW_GEOMETRY_SUFFIX).size(1).fetchSource(includes, null);
+            for (String field : Arrays.asList(aggregationModel.rawGeometries.sort.split(","))) {
+                String unsignedField = (field.startsWith("+") || field.startsWith("-")) ? field.substring(1) : field;
+                ElasticTool.checkAliasMappingFields(client, collectionReference.params.indexName, collectionReference.params.typeName, unsignedField);
+                if (field.startsWith("+")) {
+                    topHitsAggregationBuilder.sort(unsignedField, SortOrder.ASC);
+                } else if (field.startsWith("-")) {
```

Comment:
if (field.startsWith("-")) ...
else ...
to avoid an `if`?

Reply:
fixed